pkg/tmpl/sanitizer.go

```diff
@@ -183,18 +205,18 @@ func normalizeURLs(s string, settings *print.Settings) string {
 	return s
 }
 
-// generateIndentation generates indentation of Markdown headers
-// with base level of provided 'settings.MarkdownIndent' plus any
+// generateIndentation generates indentation of Markdown and AsciiDoc headers
+// with base level of provided 'settings.IndentLevel' plus any
 // extra level needed for subsection (e.g. 'Required Inputs' which
 // is a subsection of 'Inputs' section)
-func generateIndentation(extra int, settings *print.Settings) string {
-	var base = settings.MarkdownIndent
+func generateIndentation(extra int, settings *print.Settings, char string) string {
+	var base = settings.IndentLevel
 	if base < 1 || base > 5 {
 		base = 2
 	}
 	var indent string
 	for i := 0; i < base+extra; i++ {
-		indent += "#"
+		indent += char
```

Comment:
What happens if `char` is empty? What do you think we error out if that happens or may be just return an empty string?

Reply:
The error is: `Error: template: requirements:3:3: executing "requirements" at <indent>: wrong number of args for indent: want 2 got 1`. I think returning an empty string doesnt have sense as this is a required aregument for the indent function.

Reply:
That happens when you don't pass any extra arg to indent, `{{ indent 0 }}` which is wrong and we don't need to do something for this, my question was for `{{ indent 0 "" }}`, which returns and empty string. So for having a better performance I suggest to do something like this immediately inside `generateIndentation`:

```
func generateIndentation(extra int, char string, settings *print.Settings) string {
	if char == "" {
		return ""
	}
	var base = settings.IndentLevel
	...
```